tests/testthat/test_bake.R

```diff
@@ -21,3 +21,32 @@ test_that("can use tidyselect ops in bake() and juice() column selection", {
   expect_named(x, "carb")
   expect_named(y, "carb")
 })
+
+test_that("bake errors if predictor cols are missing from new_data", {
+
+  error1 <- "The following cols are missing from `new_data`: mpg"
+  error2 <- "The following cols are missing from `new_data`: cyl"
+  error26 <- "The following cols are missing from `new_data`: cyl, disp, hp, drat, wt"
+  rec1 <- recipe(mpg ~ ., data = mtcars)
+
+  expect_error(rec1 %>% prep() %>% bake(mtcars), NA)
+  expect_error(rec1 %>% prep() %>% bake(mtcars[-1]), NA)
+  expect_error(rec1 %>% prep() %>% bake(mtcars[-2]), error2)
+  expect_error(rec1 %>% prep() %>% bake(mtcars[-(2:6)]), error26)
+
+  mtcars_mat <- as.matrix(mtcars)
+  rec2 <- recipe(mpg ~ ., data = mtcars_mat)
+
+  expect_error(rec2 %>% prep() %>% bake(mtcars_mat), NA)
+  expect_error(rec2 %>% prep() %>% bake(mtcars_mat[, -1]), NA)
+  expect_error(rec2 %>% prep() %>% bake(mtcars_mat[, -2]), error2)
+  expect_error(rec2 %>% prep() %>% bake(mtcars_mat[, -(2:6)]), error26)
+
+  mtcars_mat <- as.matrix(mtcars)
+  rec3 <- recipe(x = mtcars_mat)
+
+  expect_error(rec3 %>% prep() %>% bake(mtcars_mat), NA)
+  expect_error(rec3 %>% prep() %>% bake(mtcars_mat[, -1]), error1)
+  expect_error(rec3 %>% prep() %>% bake(mtcars_mat[, -2]), error2)
+  expect_error(rec3 %>% prep() %>% bake(mtcars_mat[, -(2:6)]), error26)
+})
```

Comment:
Doesn't this break case weights? i.e. they are required at `prep()` time but not used at `bake()` time - and im pretty sure that triggers an error?

Reply:
It might, this PR is pre case weights. So I wouldn't be surprised if it doesn't work in its current state